console/src/components/Services/Data/TableBrowseRows/ViewActions.js

```diff
@@ -113,6 +101,50 @@ const vMakeRequest = () => {
   };
 };
 
+const vMakeCountRequest = () => {
+  return (dispatch, getState) => {
+    const state = getState();
+    const url = Endpoints.query;
+    const originalTable = getState().tables.currentTable;
+    dispatch({ type: V_COUNT_REQUEST_START });
+
+    const requestBody = {
+      type: 'count',
+      args: {
+        ...state.tables.view.query,
+        table: {
+          name: state.tables.currentTable,
+          schema: getState().tables.currentSchema,
+        },
+      },
+    };
+    const options = {
+      method: 'POST',
+      body: JSON.stringify(requestBody),
+      headers: dataHeaders(getState),
+      credentials: globalCookiePolicy,
+    };
+    return dispatch(requestAction(url, options)).then(
+      data => {
+        if (originalTable === getState().tables.currentTable) {
```

Comment:
What is the reason behind this check?
Isn't `originalTable` defined as `getState().tables.currentTable` in the same scope?

Reply:
It's prev logic.

Reply:
The idea is that user may change table while request is pending. And there might be race condition with another fetch data query.

Reply:
This was buggy indeed. Fixed it now